uni/lib/view/Widgets/schedule_row.dart

```diff
@@ -3,30 +3,39 @@ import 'package:uni/view/Widgets/schedule_event_rectangle.dart';
 import 'package:uni/view/Widgets/schedule_time_interval.dart';
 import 'package:add_2_calendar/add_2_calendar.dart';
 import 'package:material_design_icons_flutter/material_design_icons_flutter.dart';
+import 'package:uni/controller/local_storage/app_shared_preferences.dart';
+import 'package:uni/model/entities/exam.dart';
 
-class ScheduleRow extends StatelessWidget {
-  final String subject;
-  final List<String> rooms;
-  final String begin;
-  final String end;
-  final DateTime date;
-  final String teacher;
-  final String type;
+class ScheduleRow extends StatefulWidget {
+  final Exam exam;
+  final List<Exam> exams;
+  final bool mainPage;
 
-  const ScheduleRow(
-      {Key? key,
-      required this.subject,
-      required this.rooms,
-      required this.begin,
-      required this.end,
-      required this.date,
-      required this.teacher,
-      required this.type})
-      : super(key: key);
+  const ScheduleRow({
+    Key? key,
+    required this.exam,
+    required this.exams,
+    required this.mainPage,
+  }) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() {
+    // ignore: no_logic_in_create_state
+    return _ScheduleRowState(exams.contains(exam), exams, mainPage);
```

Comment:
We should never ignore the lint suggestions.
By previous suggestions, the warning should disappear